entity-framework/core/miscellaneous/nativeaot-and-precompiled-queries.md

```diff
@@ -0,0 +1,110 @@
+---
+title: NativeAOT Support and Precompiled Queries (Experimental) - EF Core
+description: Publishing NativeAOT Entity Framework Core applications and using precompiled queries
+author: roji
+ms.date: 11/10/2024
+uid: core/miscellaneous/nativeaot-and-precompiled-queries
+---
+# NativeAOT Support and Precompiled Queries (Experimental)
+
+> [!WARNING]
+> NativeAOT and query precompilation are highly experimental feature, and are not yet suited for production use. The support described below should be viewed as infrastructure towards the final feature, which will likely be released with EF 10. We encourage you to experiment with the current support and report on your experiences, but recommend against deploying EF NativeAOT applications in production. See below for specific known limitations.
+
+[.NET NativeAOT](/dotnet/core/deploying/native-aot) allows publishing self-contained .NET applications that have been compiled ahead-of-time (AOT). Doing so offers the following advantages:
+
+* Significantly faster application startup time
+* Small, self-contained binaries that have smaller memory footprints and are easier to deploy
+* Running applications in environments where just-in-time compilation isn't supported
+
+EF NativeAOT applications start up much faster than the same applications without NativeAOT; aside from the general startup improvements that NativeAOT offers (i.e. no JIT compilation required on each startup), EF's NativeAOT supports removes the processing of LINQ queries and their translation to SQL; the more EF LINQ queries an application has in its code, the faster the startup gains are expected to be from NativeAOT.
```

Comment:
Rewrite this without using `NativeAOT` as much

Reply:
Good suggestion :)